src/shadowbox/model/metrics.ts

```diff
@@ -28,3 +28,15 @@ export interface DataUsageByUser {
 export interface DataUsageTimeframe {
   hours: number;
 }
+
+
+export interface TunneltimeMetrics {
+  access_key?: Map<string, number>;
+  country?: Map<string, number>;
+  asn?: Map<string, number>;
+}
+
+export interface TunneltimeQuery {
```

Comment:
We need to define a TunnelTimeRequest and TunnelTimeResponse object. Let's call it that. I'd say prefer request over query because it's more generic and aligned with the practices at Google.

Request and response are service concepts, and should not be in the domain model.

We will need to define the domain model concepts that we expose in the UI, but they must be decoupled.